src/libarena/lib.rs

```diff
@@ -31,43 +31,48 @@
        html_root_url = "http://doc.rust-lang.org/nightly/")]
 
 #![feature(alloc)]
-#![feature(box_syntax)]
 #![feature(core_intrinsics)]
-#![feature(heap_api)]
-#![feature(oom)]
-#![feature(ptr_as_ref)]
+#![feature(drain)]
 #![feature(raw)]
 #![feature(staged_api)]
 #![cfg_attr(test, feature(test))]
 
 extern crate alloc;
 
+use alloc::raw_vec::RawVec;
+
 use std::cell::{Cell, RefCell};
 use std::cmp;
 use std::intrinsics;
 use std::marker;
 use std::mem;
 use std::ptr;
-use std::rc::Rc;
-use std::rt::heap::{allocate, deallocate};
+use std::slice;
 
 // The way arena uses arrays is really deeply awful. The arrays are
 // allocated, and have capacities reserved, but the fill for the array
 // will always stay at 0.
-#[derive(Clone, PartialEq)]
 struct Chunk {
-    data: Rc<RefCell<Vec<u8>>>,
+    data: RawVec<u8>,
```

Comment:
I think this possibly needs to be `RawVec<UnsafeCell<u8>>` since there's effectively a transmute from `&Chunk` -> `&mut [u8]` in the various allocation methods; however I'm not totally sure it's needed because the Arena basically ensures non-aliased-ness.